We skip many gccgo tests by setting GCCGO to a bogus value. However,
this particular test doesn't properly handle a non-existing $GCCGO
hence we unset the environment variable here.

diff -upr go.orig/src/cmd/go/testdata/script/gccgo_link_c.txt go/src/cmd/go/testdata/script/gccgo_link_c.txt
--- go.orig/src/cmd/go/testdata/script/gccgo_link_c.txt	2024-02-07 18:56:10.472634188 +0100
+++ go/src/cmd/go/testdata/script/gccgo_link_c.txt	2024-02-07 18:56:26.679400599 +0100
@@ -5,6 +5,8 @@
 [!exec:gccgo] skip
 [cross] skip  # gccgo can't necessarily cross-compile, so don't assume it will reach the step where we expect it to fail
 
+env GCCGO=
+
 ! go build -x -compiler gccgo
 stderr 'gccgo.*\-L [^ ]*alibpath \-lalib' # make sure that Go-inline "#cgo LDFLAGS:" ("-L alibpath -lalib") passed to gccgo linking stage
 ! stderr 'gccgo.*-lalib.*-lalib' # make sure -lalib is only passed once
